src/x/xCostMatrix.ts

```diff
@@ -0,0 +1,27 @@
+import { DoubleArray } from 'cheminfo-types';
+import { Matrix } from 'ml-matrix';
+
+const absDiff = (a: number, b: number) => Math.abs(a - b);
+
+interface XCostMatrixOptions {
+  fct?: (a: number, b: number) => number;
+}
```

Comment:
When there are just 'couple' of options no need to create a type and we can define them 'inline' like in https://github.com/mljs/spectra-processing/blob/7294f9c756566d5248f4ef63a3b62e6436f32aa6/src/zones/zonesWithPoints.ts#L23-L32

Please also don't forget to add JsDoc as well as the default value for 'fct' in the documentation.